server/client.go

```diff
@@ -751,6 +776,12 @@ func (c *client) readLoop() {
 		return
 	}
 
+	defer func() {
```

Comment:
Why do we need to do here as a defer?

Reply:
because readLoop has 4 returns. Again, this is a one time event when connection is closed.

Reply:
ok

Reply:
could replace `return` statements with `break` and do this at end of loop. Hopefully people working on readLoop code won't add returns in middle (although this is not dramatic if this is not set to nil, just helps).

Reply:
Can we clear these in closeConnection (or clearConnection)?

Reply:
No, it would not be safe because these are used in readLoop without lock and so should not be cleared from any other thread.

Reply:
ok